File: zeroinstall/injector/_runenv.py
"""
Helper script for <executable> bindings.
"""

# Copyright (C) 2011, Thomas Leonard
# See the README file for details, or visit http://0install.net.

import os, sys

def main():
	envname = os.path.basename(sys.argv[0])
	import json
	args = json.loads(os.environ["0install-runenv-" + envname])
	os.execv(args[0], args + sys.argv[1:])

File: src/runenv.ml
(* Copyright (C) 2013, Thomas Leonard
 * See the README file for details, or visit http://0install.net.
 *)

(* The stand-alone 0install-runenv.exe (only used on Windows) *)

module RealSystem = Support.System.RealSystem(Unix)

let () =
  let system = new RealSystem.real_system in
  Support.Utils.handle_exceptions (Runenv_shared.runenv system) (Array.to_list Sys.argv)
